lib/virt_autotest/utils.pm

```diff
@@ -481,11 +485,21 @@ sub download_script {
             # Have to output debug info at here because no logs will be uploaded if there are connection problems
             if (script_run("ssh root\@$machine 'hostname'") == 0) {
                 $script_url =~ /^https?:\/\/([\w\.]+)(:\d+)?\/.*/;
-                script_run("ssh root\@$machine 'ping $1'");
+                record_info("Guest $machine ssh accessible from host", "Debugging its network availability", result => 'fail');
+                # Debug: to check where the access problem lies in.
+                script_run("ssh root\@$machine 'ping -c3 $1'");
                 script_run("ssh root\@$machine 'traceroute $1'");
+                script_run("ssh root\@$machine 'ip route show'");
                 script_run("ssh root\@$machine 'ping -c3 openqa.suse.de'");
+                # Debug: to check if there is problem with DNS resolution: OSD <=> 10.145.10.207
+                script_run("ssh root\@$machine 'ping -c3 10.145.10.207'");
+                # Debug: to check if the guest can access its host
+                script_run("ssh root\@$machine 'ping -c3 192.168.123.1'");
                 script_run("ssh root\@$machine 'nslookup " . get_var('WORKER_HOSTNAME', 'openqa.suse.de') . "'");
+                script_run("ssh root\@$machine 'ip a'");
                 script_run("ssh root\@$machine 'cat /etc/resolv.conf'");
+                script_run('cat /etc/resolv.conf');
```

Comment:
There are so many tests in this section. What are you trying to achieve ?

Reply:
The purpose is to output more debugging information when failures happen. This kind of problems usually occur sporadically in certain specific environments, and obtaining debug information directly after occurrence can help a lot locate the cause, which greatly saves effort compared to reproducing it later.